### ISSUE TITLE
[java] CheckResultSet false-positive with local variable checked in a while loop

### ISSUE THREAD

**Issue by jbisotti** [pmd/pmd#5198]
**Affects PMD Version:** 7.5.0 (possibly earlier 7.x versions)

**Rule:** CheckResultSet

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#checkresultset>

**Description:** rs.next() is assigned to a local variable then checked/re-assigned in a while loop.

**Code Sample demonstrating the issue:**

```java
        ...

        boolean resultSetHasNext = resultSet.next();    // Line 286

        while (resultSetHasNext && resultMap.size() < maxResultCap) {
            ...
            resultSetHasNext = resultSet.next();
        }

        ...
```

**Expected outcome:** There is no reported violation.

PMD reports a violation at line 286, but that's wrong. That's a false positive.

**Running PMD through:** *Maven*
